packages/hardware-ledger/src/LedgerKeyAgent.ts

```diff
@@ -45,7 +56,106 @@ export interface CreateLedgerTransportProps {
 }
 
 const transportTypedError = (error?: any) =>
-  new AuthenticationError('Ledger transport failed', new TransportError('Ledger transport failed', error));
+  new errors.AuthenticationError(
+    'Ledger transport failed',
+    new errors.TransportError('Ledger transport failed', error)
+  );
+
+const hasRegistrationOrRetirementCerts = (certificates: Certificate[] | null | undefined): boolean => {
+  if (!certificates) return false;
+
+  return (
+    certificates.some((cert) => cert.type === CertificateType.STAKE_POOL_RETIREMENT) ||
+    certificates.some((cert) => cert.type === CertificateType.STAKE_POOL_REGISTRATION)
+  );
+};
+
+const stakeCredentialCert = (cert: Certificate) =>
+  cert.type === CertificateType.STAKE_REGISTRATION ||
+  cert.type === CertificateType.STAKE_DEREGISTRATION ||
+  cert.type === CertificateType.STAKE_DELEGATION;
+
+interface StakeCredentialCertificateParams {
+  stakeCredential: StakeCredentialParams;
+}
+
+const containsOnlyScriptHashCreds = (tx: Transaction): boolean => {
+  const withdrawalsAllScriptHash = !tx.withdrawals?.some(
+    (withdrawal) => withdrawal.stakeCredential.type !== StakeCredentialParamsType.SCRIPT_HASH
+  );
+
+  if (tx.certificates) {
+    for (const cert of tx.certificates) {
+      if (!stakeCredentialCert(cert)) return false;
+
+      const certParams = cert.params as unknown as StakeCredentialCertificateParams;
+      if (certParams.stakeCredential.type !== StakeCredentialParamsType.SCRIPT_HASH) return false;
+    }
+  }
+
+  return withdrawalsAllScriptHash;
+};
+
+const isMultiSig = (tx: Transaction): boolean => {
+  const result = false;
+
+  const allThirdPartyInputs = !tx.inputs.some((input) => input.path !== null);
+  const allThirdPartyOutputs = !tx.outputs.some((out) => out.destination.type !== TxOutputDestinationType.THIRD_PARTY);
+
+  if (
+    allThirdPartyInputs &&
+    allThirdPartyOutputs &&
+    !tx.collateralInputs &&
+    !tx.requiredSigners &&
+    !hasRegistrationOrRetirementCerts(tx.certificates) &&
+    containsOnlyScriptHashCreds(tx)
+  ) {
+    return true;
+  }
+
+  return result;
+};
+
+/**
+ * Gets the mode in which we want to sign the transaction.
+ * Ledger has certain limitations due to which it cannot sign arbitrary combination of all transaction features.
```

Comment:
How unsupported combinations handled?

Reply:
@AngelCastilloB can you add some tests that demonstrate the errors we expect?